specification/purview/Azure.Analytics.Purview.DataMap/routes.tsp

```diff
@@ -0,0 +1,2245 @@
+
+NOTE:
+The attrName should be an unique attribute for the given entity-type.
+The REST
+request would look something like this
+
+GET
+/v2/entity/bulk/uniqueAttribute/type/hive_db?attr_1:qualifiedName=db1@cl1&attr_2:qualifiedName=db2@cl1
+
+Note:
+at least one unique attribute must be provided.
+""")
+  @route("/atlas/v2/entity/bulk/uniqueAttribute/type/{typeName}")
+  @get
+  listByUniqueAttributes is Azure.Core.Foundations.Operation<
+    {
+      @doc("The name of the type.")
+      @path
+      typeName: string;
+
+      @doc("Whether to return minimal information for referred entities.")
+      @query
+      minExtInfo?: boolean;
+
+      @doc("Whether to ignore relationship attributes.")
+      @query
+      ignoreRelationships?: boolean;
+
+      @doc("""
+Qualified name of an entity. E.g. to find 2 entities you can set
+attrs_1:qualifiedName=db1@cl1&attrs_2:qualifiedName=db2@cl1. (This is only an
+example. qualifiedName can be changed to other unique attributes)
+""")
+      @query
+      `attr_N:qualifiedName`?: string;
+    },
+    AtlasEntitiesWithExtInfo,
+    {},
+    AtlasErrorResponse
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "This is Atlas API"
+  @doc("Get entity header given its GUID.")
+  @route("/atlas/v2/entity/guid/{guid}/header")
+  @get
+  getHeader is Azure.Core.Foundations.Operation<
+    {
+      @doc("The globally unique identifier of the entity.")
+      @path
+      guid: string;
+    },
+    AtlasEntityHeader,
+    {},
+    AtlasErrorResponse
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "This is Atlas API"
+  @doc("Remove business metadata from an entity.")
+  @route("/atlas/v2/entity/guid/{guid}/businessmetadata")
+  @delete
+  removeBusinessMetadata is Azure.Core.Foundations.Operation<
+    {
+      @doc("The globally unique identifier of the entity.")
+      @path
+      guid: string;
+
+      #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Should use unknown to model Object"
+      @doc("Business metadata payload")
+      @body
+      body: Record<Record<unknown>>;
+    },
+    void,
+    {},
+    AtlasErrorResponse
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "This is Atlas API"
+  @doc("Add business metadata to an entity.")
+  @route("/atlas/v2/entity/guid/{guid}/businessmetadata")
+  @post
+  addOrUpdateBusinessMetadata is Azure.Core.Foundations.Operation<
+    {
+      @doc("The globally unique identifier of the entity.")
+      @path
+      guid: string;
+
+      @doc("""
+Whether to overwrite the existing business metadata on the entity or not,
+default is false.
+""")
+      @query("isOverwrite")
+      overwrite?: boolean;
+
+      #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Should use unknown to model Object"
+      @doc("BusinessMetadata payload")
+      @body
+      body: Record<Record<unknown>>;
```

Comment:
Based on the original swagger, this type shouldnt be nested. I saw that the BusinessMetadata payload was previously defined like this:
``` 
"BusinessMetadata": {
      "type": "object",
      "additionalProperties": {
        "type": "object"
      },
      "description": "BusinessMetadata"
    },
```

So this type should just be `Record<unknown>`, there are other instances of this nested Record type that should be fixed if that's the case

Reply:
For buisnessMetadata, the format is <String, Object>. This is also an Atlas concept. We try to model it as a strong type but fail to achieve that.

Reply:
I think `Record<Record<unknown>` is correct in this case. It was previously scoped to `type: object` and `additionalProperties` is true by default (now we have not really been respecting that in autorest)

Reply:
i think it depends on what will cause the breaking change in the SDK. The previous swagger might have been interpreted as `REcord<unknown>`

Reply:
It's a new SDK so if this is an improvement in the modeling then it's better to go with the updated type if that's what the service returns.

Reply:
This is a bug in the original swagger. The businessMetadata should be defined as Map<String, Map<String, Object>> instead of Map<String,Object>

Reply:
Although on line 736 the business metadata payload is typed as `Record<unknown>`, it looks like it's for the same type of response, is that correct?

Reply:
Yes. It is correct. the 736 is business metadata **attributes**, which is Map<String,Object>